src/ORM/DataList.php

```diff
@@ -1252,4 +1252,27 @@ public function offsetUnset($key)
     {
         user_error("Can't alter items in a DataList using array-access", E_USER_ERROR);
     }
+
+    /**
+     * Iterate over this datalist in "chunks". This will break the query in smaller subset and avoid loading the entire
+     * result set in memory at once. Beware not to perform any operations on the results that might alter the return
+     * order. Otherwise, you might break subsequent chunks.
+     * @param int $chunkSize
+     * @return Generator|DataObject[]
+     */
+    public function chunk($chunkSize = 100)
+    {
+        if ($chunkSize < 1) {
+            throw new InvalidArgumentException('DataList::chunk: chunksize must be greater than or equal to 1');
+        }
+
+        $count = $this->count();
```

Comment:
I feel like you could implement this without performing a count - which would save a potentially expensive query. For example counting a DataList while using Versioned in archive mode isn't necessarily fast.

You could do something like:
```
while ($chunk = $this->limit($chunkSize, $chunkSize * $i)->execute()) {
	if (!$chunk->count()) { return; } // This should perform just a PHP count because of the execute call
    yield $chunk;
}
```

This is also a subtle suggestion to have better naming than `$do`.

Reply:
Won't I end up with an extra query any way at the end? Let's say I have 20 results and I'm chunking my query by 10. After I get my second chunk, the `while` loop will run again and discover there's no results after offset 20.

Reply:
Good point, you would - but this query for non-existing rows should be faster - and it's also at the end so code that might not actually need all chunks won't be affected.

Reply:
Can probably do somehting like this, which should only do the extra query when the result set count is a multiple of `$chunkSize`.
```php
while ($chunk = $this->limit($chunkSize, $chunkSize * $page)->execute()) {
    foreach ($chunk as $item) {
        yield $file;
    }

    if ($chunk->count() < $chunkSize) {
        break;
    }

    $page++;
}
```

Reply:
The logic on that last comment has been applied.